JS-Basics/index.js

```diff
@@ -444,4 +444,89 @@ function isContainsAllCharactersInT(s,t) {
 
 }
 
-console.log(minimumWindowSubString);
\ No newline at end of file
+console.log(minimumWindowSubString);
+
+
+// 6. Rotting Oranges
+let grid = [[2, 1, 1], [1, 1, 0], [0, 1, 1]];
+let duplicate = structuredClone(grid);
```

Comment:
1. We can clone by reference with the following syntax which create another reference to the same object. Any change made in _user_ is also reflected in _admin_ object.

> let admin = user;

2. Object.assign() method can be used to create a shallow copy were the destination and source objects will point to two different references, but this will not work for nested object.

> let clone = Object.assign({}, user);

3. Deep copy also works for nested object i.e. another object as the value for a property in an object.

> let clone = structuredClone(user);

Reply:
Can spread operater be used